repository/src/raw/tests.rs

```diff
@@ -448,3 +450,25 @@ async fn remote() {
         )]
     );
 }
+
+#[tokio::test]
+async fn reserved_state() {
+    let td = TempDir::new().unwrap();
+    let path = td.path();
+    let mut repo = init_repository_with_initial_commit(path).await.unwrap();
+
+    let (rs, _) = simperby_test_suite::generate_standard_genesis(10);
+
+    repo.checkout(MAIN.into()).await.unwrap();
```

Comment:
repo currently is in "main" branch, so I don't think this line is needed.

Reply:
that is true but its not trivial. (it's highly dependent on the implementation of `init_repository_with_initial_commit`).

Reply:
Agree